README.md
# XMSegmentedControl

[![Version](https://img.shields.io/cocoapods/v/XMSegmentedControl.svg?style=flat)](http://cocoapods.org/pods/XMSegmentedControl)
[![License](https://img.shields.io/cocoapods/l/XMSegmentedControl.svg?style=flat)](http://cocoapods.org/pods/XMSegmentedControl)
[![Platform](https://img.shields.io/cocoapods/p/XMSegmentedControl.svg?style=flat)](http://cocoapods.org/pods/XMSegmentedControl)

## Overview
XMSegmentedControl is a customizable segmented control. It allows using Text, Icons, or a combination of Icons and Text as the segments.

![Screenshot](http://zippy.gfycat.com/SlimZestyGermanspitz.gif)

## Requirements
* iOS 8+

## Installation
You can install XMSegmentedControl manually by simply copying `XMSegmentedControl.swift` into your project's workspace.  

XMSegmentedControl is available through [CocoaPods](http://cocoapods.org).  To install it, simply add the following line to your Podfile:

```ruby
pod "XMSegmentedControl"
```
## Usage

* Import XMSegmentedControl into your project. 

```Swift
import XMSegmentedControl
```
* Use the Storyboard and drag a UIView. 
* Select the view and under the Identity Inspector select `XMSegmentedControl` under class. 
* Conform to the `XMSegmentedControlDelegate` protocol by implementing `func xmSegmentedControl(xmSegmentedControl: XMSegmentedControl, selectedSegment: Int)` to receive notifications on which segment was selected.

```Swift
func xmSegmentedControl(xmSegmentedControl: XMSegmentedControl, selectedSegment: Int) {
print("SegmentedControl Selected Segment: \(selectedSegment)")
}
```
Alternatively, create a XMSegmentedControl from code.

### Text Only Segmented Control
The example below creates a text-only segmented control.
```Swift
let segmentedControl3 = XMSegmentedControl(frame: CGRect(x: 0, y: 164, width: self.view.frame.width, height: 44), segmentTitle: ["Hello", "World", "Three"], selectedItemHighlightStyle: XMSelectedItemHighlightStyle.TopEdge)

segmentedControl3.backgroundColor = UIColor(red: 22/255, green: 150/255, blue: 122/255, alpha: 1)
segmentedControl3.highlightColor = UIColor(red: 25/255, green: 180/255, blue: 145/255, alpha: 1)
segmentedControl3.tint = UIColor.whiteColor()
segmentedControl3.highlightTint = UIColor.blackColor()

self.view.addSubview(segmentedControl3)
```

![TextOnly](https://dl.dropboxusercontent.com/u/72507896/XMSegmentedControlScreenshots/textonly.png)

### Icon Only Segmented Control
The example below creates an icon-only segmented control.
``` Swift
let icons1:[UIImage] = [UIImage(named: "icon1")!, UIImage(named: "icon2")!, UIImage(named: "icon3")!, UIImage(named: "icon4")!, UIImage(named: "icon5")!, UIImage(named: "icon6")!]

let segmentedControl4 = XMSegmentedControl(frame: CGRect(x: 0, y: 214, width: self.view.frame.width, height: 44), segmentIcon: icons1, selectedItemHighlightStyle: XMSelectedItemHighlightStyle.Background)

segmentedControl4.backgroundColor = UIColor(red: 128/255, green: 59/255, blue: 159/255, alpha: 1)
segmentedControl4.highlightColor = UIColor(red: 144/255, green: 79/255, blue: 173/255, alpha: 1)
segmentedControl4.tint = UIColor.whiteColor()

self.view.addSubview(segmentedControl4)
```

![IconOnly](https://dl.dropboxusercontent.com/u/72507896/XMSegmentedControlScreenshots/icononly.png)

### Icon + Text Segmented Control
The example below creates an Icon + Text segmented control.
``` Swift
let backgroundColor = UIColor(red: 205/255, green: 74/255, blue: 1/255, alpha: 1)
let highlightColor = UIColor(red: 226/255, green: 114/255, blue: 31/255, alpha: 1)

let titles = ["Hello", "World", "Two"]
let icons = [UIImage(named: "icon1")!, UIImage(named: "icon2")!, UIImage(named: "icon3")!]
let frame = CGRect(x: 0, y: 114, width: self.view.frame.width, height: 44)

let segmentedControl2 = XMSegmentedControl(frame: frame, segmentContent: (titles, icons), selectedItemHighlightStyle: XMSelectedItemHighlightStyle.BottomEdge)

segmentedControl2.backgroundColor = backgroundColor
segmentedControl2.highlightColor = highlightColor
segmentedControl2.tint = UIColor.whiteColor()
segmentedControl2.highlightTint = UIColor.blackColor()

self.view.addSubview(segmentedControl2)
```

![IconText](https://dl.dropboxusercontent.com/u/72507896/XMSegmentedControlScreenshots/texticon.tiff)

For more examples see the Example Project provided.

## Example Project

To run the example project, clone the repo, and run `pod install` from the Example directory first.
Icons in the example project are provided by http://icons8.com

## Author

Xavier Merino, xaviermerino@gmail.com

## License

XMSegmentedControl is available under the MIT license. See the LICENSE file for more info.

Example/CHANGELOG.md
#Change Log
All notable changes to this project will be documented in this file.
`XMSegmentedControl` adheres to [Semantic Versioning](http://semver.org/).

--- 

##0.1.3 (11/28/2015)

#### Added
* Hybrid mode: Allows using a combination of text and icons for a tab.

####Fixed
* Fixed an issue where device rotation would reset the segmented control.

---

##0.1.0 (10/1/2015)

First Release

Example/Tests/Tests.swift
import UIKit
import XCTest
import XMSegmentedControl

class Tests: XCTestCase {
    
    override func setUp() {
        super.setUp()
        // Put setup code here. This method is called before the invocation of each test method in the class.
    }
    
    override func tearDown() {
        // Put teardown code here. This method is called after the invocation of each test method in the class.
        super.tearDown()
    }
    
    func testExample() {
        // This is an example of a functional test case.
        XCTAssert(true, "Pass")
    }
    
    func testPerformanceExample() {
        // This is an example of a performance test case.
        self.measureBlock() {
            // Put the code you want to measure the time of here.
        }
    }
    
}


Pod/Classes/XMSegmentedControl.swift
/*

Copyright © 2015 Xavier Merino. All rights reserved.

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
THE SOFTWARE.

*/

//
//  XMSegmentedControl.swift
//  XMSegmentedControl
//
//  Created by Xavier Merino on 9/29/15.
//  Updated by Xavier Merino on 11/28/15.
//

import UIKit

///The delegate of `XMSegmentedControl` must adopt `XMSegmentedControlDelegate` protocol. It allows retrieving information on which segment was tapped.
public protocol XMSegmentedControlDelegate {
    /// Tells the delegate that a specific segment is now selected.
    func xmSegmentedControl(xmSegmentedControl: XMSegmentedControl, selectedSegment:Int)
}

/**
 Highlighted Styles for the selected segments.
 - Background: The background of the selected segment is highlighted.
 - TopEdge: The top edge of the selected segment is highlighted.
 - BottomEdge: The bottom edge of the selected segment is highlighted.
 */
public enum XMSelectedItemHighlightStyle {
    case Background
    case TopEdge
    case BottomEdge
}

/**
 Content Type for the segmented control.
 - Text: The segmented control displays only text.
 - Icon: The segmented control displays only icons/images.
 - Hybrid: The segmented control displays icons and text.
 */
public enum XMContentType {
    case Text
    case Icon
    case Hybrid
}

/**
 XMSegmentedControl is a replacement for the standard segmented control. 
 It allows text, icon, and text + icon segments.
 Conform to `XMSegmentedControlDelegate` to obtain the selected segment.
 */

@IBDesignable
public class XMSegmentedControl: UIView {
    public var delegate:XMSegmentedControlDelegate?
    private var highlightView:UIView!
    
    /**
     Defines the height of the highlighted edge if `selectedItemHighlightStyle` is either `TopEdge` or `BottomEdge`
     - Note: Changes only take place if `selectedItemHighlightStyle` is either `TopEdge` or `BottomEdge`
     */
    public var edgeHighlightHeight:CGFloat = 5.0
    
    /// Changes the background of the selected segment.
    @IBInspectable public var highlightColor:UIColor = UIColor(red: 42/255, green: 132/255, blue: 210/255, alpha: 1) {
        didSet {
            update()
        }
    }
    
    /// Changes the font color or the icon tint color for the segments.
    @IBInspectable public var tint:UIColor = UIColor.whiteColor() {
        didSet {
            update()
        }
    }
    
    /// Changes the font color or the icon tint for the selected segment.
    @IBInspectable public var highlightTint:UIColor = UIColor.whiteColor() {
        didSet {
            update()
        }
    }
    
    /**
     Sets the segmented control content type to `Text` and uses the content of the array to create the segments.
     - Note: Only six elements will be displayed.
     */
    public var segmentTitle:[String] = []{
        didSet {
            segmentTitle = segmentTitle.count > 6 ? Array(segmentTitle[0..<6]) : segmentTitle
            contentType = .Text
            update()
        }
    }
    
    /**
     Sets the segmented control content type to `Icon` and uses the content of the array to create the segments.
     - Note: Only six elements will be displayed.
     */
    public var segmentIcon:[UIImage] = []{
        didSet {
            segmentIcon = segmentIcon.count > 6 ? Array(segmentIcon[0..<6]) : segmentIcon
            contentType = .Icon
            update()
        }
    }
    
    /**
     Sets the segmented control content type to `Hybrid` (i.e. displaying icons and text) and uses the content of the tuple to create the segments.
     - Note: Only six elements will be displayed. The arrays will be truncated.
     */
    public var segmentContent:(text:[String], icon:[UIImage]) = ([], []) {
        didSet {
            /// Checks if `segmentContent.text` and `segmentContent.icon` have the same size. Size must be synchronized.
            if (segmentContent.text.count == segmentContent.icon.count) {
                segmentContent.text = segmentContent.text.count > 6 ? Array(segmentContent.text[0..<6]) : segmentContent.text
                segmentContent.icon = segmentContent.icon.count > 6 ? Array(segmentContent.icon[0..<6]) : segmentContent.icon
                /// Resize images to fit within the button. See the `resizeImage` function for more details.
                segmentContent.icon = segmentContent.icon.map(resizeImage)
                contentType = .Hybrid
                update()
            } else {
                print("Text and Icon arrays out of sync.")
            }
        }
    }
    
    /**
    The index of the selected segment.
    - Note: Read-only. Changing `selectedSegment` won't do anything.
    */
    public var selectedSegment:Int = 0
    
    /**
     Sets the font for the text displayed in the segmented control.
     - Note: Changes only take place if `contentType` is `Text` or `Hybrid`
     */
    public var font:UIFont = UIFont(name: "AvenirNext-DemiBold", size: 15)!
    
    /// Sets the segmented control selected item highlight style to `Background`, `TopEdge` or `BottomEdge`.
    public var selectedItemHighlightStyle:XMSelectedItemHighlightStyle = .Background
    
    /// Sets the segmented control content type to `Text`, `Icon`, or `Hybrid`
    public var contentType:XMContentType = .Text
    
    /// Initializes and returns a newly allocated XMSegmentedControl object with the specified frame rectangle. It sets the segments of the control from the given `segmentTitle` array and the highlight style for the selected item.
    public init (frame: CGRect, segmentTitle: [String], selectedItemHighlightStyle:XMSelectedItemHighlightStyle) {
        super.init (frame: frame)
        commonInit(segmentTitle, highlightStyle: selectedItemHighlightStyle)
    }
    
    /// Initializes and returns a newly allocated XMSegmentedControl object with the specified frame rectangle. It sets the segments of the control from the given `segmentIcon` array and the highlight style for the selected item.
    public init (frame: CGRect, segmentIcon: [UIImage], selectedItemHighlightStyle:XMSelectedItemHighlightStyle) {
        super.init (frame: frame)
        commonInit(segmentIcon, highlightStyle: selectedItemHighlightStyle)
    }
    
    /// Initializes and returns a newly allocated XMSegmentedControl object with the specified frame rectangle. It sets the segments of the control from the given `segmentContent` tuple and the highlight style for the selected item. Notice that the tuple consists of an array containing the titles and another array containing the icons. The two arrays must be the same size.
    public init (frame: CGRect, segmentContent: ([String], [UIImage]), selectedItemHighlightStyle:XMSelectedItemHighlightStyle) {
        super.init (frame: frame)
        commonInit(segmentContent, highlightStyle: selectedItemHighlightStyle)
    }
    
    /// Common initializer. The `Any` type allows accepting the tuple and the array data structures. It then gets unwrapped.
    private func commonInit(data:Any, highlightStyle:XMSelectedItemHighlightStyle){
        if let segmentTitle = data as? [String] {
            self.segmentTitle = segmentTitle
        } else if let segmentIcon = data as? [UIImage] {
            self.segmentIcon = segmentIcon
        } else if let segmentContent = data as? ([String], [UIImage]) {
            self.segmentContent = segmentContent
        }
        self.backgroundColor = UIColor(red: 45/255, green: 62/255, blue: 100/255, alpha: 1)
        self.selectedItemHighlightStyle = highlightStyle
    }
    
    public override init(frame: CGRect) {
        super.init(frame: frame)
        self.backgroundColor = UIColor(red: 45/255, green: 62/255, blue: 100/255, alpha: 1)
    }
    
    required public init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        self.backgroundColor = UIColor(red: 45/255, green: 62/255, blue: 100/255, alpha: 1)
    }
    
    /// Prepares the render of the view for the Storyboard.
    override public func prepareForInterfaceBuilder() {
        segmentTitle = ["Only", "For", "Show"]
        self.backgroundColor = UIColor(red: 45/255, green: 62/255, blue: 100/255, alpha: 1)
    }
    
    /// Function gets automatically called whenever orientation changes and when view is loaded.
    override public func layoutSubviews() {
        update()
    }
    
    /**
        Forces the segmented control to reload. 
        Function gets called whenever orientation is changed.
    */
    public func update(){
        
        /**
         Adds the segments to the segmented control. It takes `starting` where you specify the starting X position for the first segment. `sections` takes the number of total segments to be added. `width` and `height` specify size constraints for the segment.
         */
        func addSegments(startingPosition starting:CGFloat, sections:Int, width:CGFloat, height:CGFloat){
            for (var i = 0; i < sections; i++){
                let frame = CGRect(x: starting + (CGFloat(i) * width), y: 0, width: width, height: height)
                let tab:UIButton = UIButton(type: UIButtonType.System)
                tab.frame = frame
                
                switch contentType {
                case .Icon:
                    /// Icon Mode sets insets. In a 44 height control the image becomes 20x20. The imageView in the UIButton is set to ScaleAspectFit to preserve aspect ratio and resize to fit.
                    tab.imageEdgeInsets = UIEdgeInsets(top: 12, left: 12, bottom: 12, right: 12)
                    tab.imageView?.contentMode = UIViewContentMode.ScaleAspectFit
                    tab.tintColor = i == selectedSegment ? highlightTint : tint
                    tab.setImage(segmentIcon[i], forState: .Normal)
                    
                case .Text:
                    tab.setTitle(segmentTitle[i], forState: .Normal)
                    tab.setTitleColor(i == selectedSegment ? highlightTint : tint, forState: .Normal)
                    tab.titleLabel?.font = font
                    
                case .Hybrid:
                    /// Modify `insetAmount` to change spacing between borders, icon, and text
                    /// Hybrid Mode sets insets. In a 44 height control the image becomes 20x20. The imageView in the UIButton is set to ScaleAspectFit to preserve aspect ratio and resize to fit.
                    let insetAmount:CGFloat = 8 / 2.0
                    tab.imageEdgeInsets = UIEdgeInsetsMake(12, -insetAmount, 12, insetAmount)
                    tab.titleEdgeInsets = UIEdgeInsetsMake(0, insetAmount*2, 0, 0)
                    tab.contentEdgeInsets = UIEdgeInsetsMake(0, insetAmount, 0, insetAmount)
                    tab.contentHorizontalAlignment = UIControlContentHorizontalAlignment.Center
                    tab.setTitle(segmentContent.text[i], forState: .Normal)
                    tab.setImage(segmentContent.icon[i], forState: .Normal)
                    tab.titleLabel?.font = font
                    tab.imageView?.contentMode = UIViewContentMode.ScaleAspectFit
                    tab.tintColor = i == selectedSegment ? highlightTint : tint
                    
                }
                
                /// The tag identifies the index of the segment.
                tab.tag = i
                tab.addTarget(self, action: "segmentPressed:", forControlEvents: UIControlEvents.TouchUpInside)
                self.addSubview(tab)
            }
        }
        
        /**
         Adds the highlight view. The X position is specified by `starting`. The height is is set to `edgeHighlightHeight` unless you specifiy `selectedItemHighlightStyle` as `Background`
         */
        func addHighlightView(startingPosition starting:CGFloat, width:CGFloat){
            switch selectedItemHighlightStyle {
            case .Background:
                highlightView = UIView(frame: CGRect(x: starting, y: 0, width: width, height: self.frame.height))
            case .TopEdge:
                highlightView = UIView(frame: CGRect(x: starting, y: 0, width: width, height: edgeHighlightHeight))
            case .BottomEdge:
                highlightView = UIView(frame: CGRect(x: starting, y: self.frame.height - edgeHighlightHeight, width: width, height: edgeHighlightHeight))
            }
            highlightView.backgroundColor = highlightColor
            self.addSubview(highlightView)
        }
        
        /// Removes all segments from the segment control.
        (self.subviews as [UIView]).forEach { $0.removeFromSuperview() }
        let totalWidth = self.frame.width
        
        if contentType == .Text {
            let tabBarSections:Int = segmentTitle.count
            let sectionWidth = totalWidth / CGFloat(tabBarSections)
            addHighlightView(startingPosition: CGFloat(selectedSegment) * sectionWidth, width: sectionWidth)
            addSegments(startingPosition: 0, sections: tabBarSections, width: sectionWidth, height: self.frame.height)
        } else if contentType == .Icon {
            /// Icon Mode assumes that there are always 6 segments so that it can center the icons in the middle of the bar.
            let tabBarSections:Int = segmentIcon.count
            let sectionWidth = totalWidth / 6
            let availableSpace = totalWidth - (sectionWidth * CGFloat(6 - tabBarSections))
            let startingXPosition = (totalWidth - availableSpace) / 2
            addHighlightView(startingPosition: startingXPosition + (sectionWidth * CGFloat(selectedSegment)), width: sectionWidth)
            addSegments(startingPosition: startingXPosition, sections: tabBarSections, width: sectionWidth, height: self.frame.height)
        } else { // Hybrid
            let tabBarSections:Int = segmentContent.text.count
            let sectionWidth = totalWidth / CGFloat(tabBarSections)
            addHighlightView(startingPosition: CGFloat(selectedSegment) * sectionWidth, width: sectionWidth)
            addSegments(startingPosition: 0, sections: tabBarSections, width: sectionWidth, height: self.frame.height)
        }
    }
    
    /// Called whenever a segment is pressed. Sends the information to the delegate.
    @objc private func segmentPressed(sender:UIButton){
        /// Returns true if the view is a `UIButton`
        func isUIButton(view:UIView) -> Bool {
            return view is UIButton ? true : false
        }

        let xPosition:CGFloat = sender.frame.origin.x
        let newPosition:CGPoint = CGPoint(x: xPosition, y: highlightView.frame.origin.y)
        
        /// Animates the movement of the `highlightView` and sets the appropiate tint.
        UIView.animateWithDuration(0.3, delay: 0, usingSpringWithDamping: 0.7, initialSpringVelocity: 0.7, options: UIViewAnimationOptions.CurveEaseOut, animations: {
            self.highlightView.frame.origin = newPosition
            
            switch(self.contentType){
            case .Icon, .Hybrid:
                ((self.subviews.filter(isUIButton)) as! [UIButton]).forEach { $0.tintColor = self.tint }
                sender.tintColor = self.highlightTint
            case .Text:
                ((self.subviews.filter(isUIButton)) as! [UIButton]).forEach { $0.setTitleColor(self.tint, forState: .Normal) }
                sender.setTitleColor(self.highlightTint, forState: .Normal)
            }
            
            }, completion: nil)
        
        selectedSegment = sender.tag
        delegate?.xmSegmentedControl(self, selectedSegment: selectedSegment)
    }
    
    /**
        Scales an image to the set size. The size is 20x20. It takes into account alpha and uses the screen's scale to resize. 
        - ToDo: Allow the size to be variable and changed instead of fixed. As of now it looks good in a segmentedControl with a height of 44.
    */
    private func resizeImage(image:UIImage) -> UIImage {
        let size = CGSize(width: 20, height: 20)
        UIGraphicsBeginImageContextWithOptions(size, false, 0)
        image.drawInRect(CGRect(origin: CGPointZero, size: size))
        let scaledImage = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()
        return scaledImage
    }
}

Example/XMSegmentedControl/AppDelegate.swift
//
//  AppDelegate.swift
//  XMSegmentedControl
//
//  Created by xaviermerino on 10/01/2015.
//  Copyright (c) 2015 xaviermerino. All rights reserved.
//

import UIKit

@UIApplicationMain
class AppDelegate: UIResponder, UIApplicationDelegate {

    var window: UIWindow?


    func application(application: UIApplication, didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool {
        // Override point for customization after application launch.
        return true
    }

    func applicationWillResignActive(application: UIApplication) {
        // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
        // Use this method to pause ongoing tasks, disable timers, and throttle down OpenGL ES frame rates. Games should use this method to pause the game.
    }

    func applicationDidEnterBackground(application: UIApplication) {
        // Use this method to release shared resources, save user data, invalidate timers, and store enough application state information to restore your application to its current state in case it is terminated later.
        // If your application supports background execution, this method is called instead of applicationWillTerminate: when the user quits.
    }

    func applicationWillEnterForeground(application: UIApplication) {
        // Called as part of the transition from the background to the inactive state; here you can undo many of the changes made on entering the background.
    }

    func applicationDidBecomeActive(application: UIApplication) {
        // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
    }

    func applicationWillTerminate(application: UIApplication) {
        // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
    }


}



Example/XMSegmentedControl/ViewController.swift
//
//  ViewController.swift
//  XMSegmentedControl
//
//  Created by xaviermerino on 10/01/2015.
//  Copyright (c) 2015 Xavier Merino. All rights reserved.
//

import UIKit
import XMSegmentedControl

class ViewController: UIViewController, XMSegmentedControlDelegate {

    @IBOutlet weak var segmentedControl1: XMSegmentedControl!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        /*
            Initialized from Storyboard
        */
        
        segmentedControl1.delegate = self
        segmentedControl1.segmentTitle = ["Hello", "World"]
        
        /*
            Examples created from code
            The segmented control can only display up to six elements. 
            It will ignore the excess.
        */
        
        // SegmentedControl2
        
        let backgroundColor = UIColor(red: 205/255, green: 74/255, blue: 1/255, alpha: 1)
        let highlightColor = UIColor(red: 226/255, green: 114/255, blue: 31/255, alpha: 1)
        
        let titles = ["Hello", "World", "Two"]
        let icons = [UIImage(named: "icon1")!, UIImage(named: "icon2")!, UIImage(named: "icon3")!]
        
        let frame = CGRect(x: 0, y: 114, width: self.view.frame.width, height: 44)
        
        let segmentedControl2 = XMSegmentedControl(frame: frame, segmentContent: (titles, icons), selectedItemHighlightStyle: XMSelectedItemHighlightStyle.BottomEdge)
        segmentedControl2.backgroundColor = backgroundColor
        segmentedControl2.highlightColor = highlightColor
        segmentedControl2.tint = UIColor.whiteColor()
        segmentedControl2.highlightTint = UIColor.blackColor()
        
        self.view.addSubview(segmentedControl2)
        
        // SegmentedControl3
        
        let segmentedControl3 = XMSegmentedControl(frame: CGRect(x: 0, y: 164, width: self.view.frame.width, height: 44), segmentTitle: ["Hello", "World", "Three"], selectedItemHighlightStyle: XMSelectedItemHighlightStyle.TopEdge)
        segmentedControl3.backgroundColor = UIColor(red: 22/255, green: 150/255, blue: 122/255, alpha: 1)
        segmentedControl3.highlightColor = UIColor(red: 25/255, green: 180/255, blue: 145/255, alpha: 1)
        segmentedControl3.tint = UIColor.whiteColor()
        segmentedControl3.highlightTint = UIColor.blackColor()
        self.view.addSubview(segmentedControl3)
        
        // SegmentedControl4
        
        let icons1:[UIImage] = [UIImage(named: "icon1")!, UIImage(named: "icon2")!, UIImage(named: "icon3")!, UIImage(named: "icon4")!, UIImage(named: "icon5")!, UIImage(named: "icon6")!]
        let segmentedControl4 = XMSegmentedControl(frame: CGRect(x: 0, y: 214, width: self.view.frame.width, height: 44), segmentIcon: icons1, selectedItemHighlightStyle: XMSelectedItemHighlightStyle.Background)
        segmentedControl4.backgroundColor = UIColor(red: 128/255, green: 59/255, blue: 159/255, alpha: 1)
        segmentedControl4.highlightColor = UIColor(red: 144/255, green: 79/255, blue: 173/255, alpha: 1)
        segmentedControl4.tint = UIColor.whiteColor()
        self.view.addSubview(segmentedControl4)
        
        // SegmentedControl5
        
        let icons2:[UIImage] = [UIImage(named: "icon2")!, UIImage(named: "icon3")!, UIImage(named: "icon4")!, UIImage(named: "icon5")!, UIImage(named: "icon6")!, UIImage(named: "icon7")!, UIImage(named: "icon1")!]
        let segmentedControl5 = XMSegmentedControl(frame: CGRect(x: 0, y: 264, width: self.view.frame.width, height: 44), segmentIcon: icons2, selectedItemHighlightStyle: XMSelectedItemHighlightStyle.BottomEdge)
        segmentedControl5.backgroundColor = UIColor(red: 184/255, green: 50/255, blue: 38/255, alpha: 1)
        segmentedControl5.highlightColor = UIColor(red: 228/255, green: 66/255, blue: 53/255, alpha: 1)
        segmentedControl5.tint = UIColor.whiteColor()
        self.view.addSubview(segmentedControl5)
        
        // SegmentedControl6
        
        let icons3:[UIImage] = [UIImage(named: "icon2")!, UIImage(named: "icon3")!, UIImage(named: "icon4")!, UIImage(named: "icon5")!, UIImage(named: "icon6")!]
        let segmentedControl6 = XMSegmentedControl(frame: CGRect(x: 0, y: 314, width: self.view.frame.width, height: 44), segmentIcon: icons3, selectedItemHighlightStyle: XMSelectedItemHighlightStyle.TopEdge)
        segmentedControl6.backgroundColor = UIColor(red: 35/255, green: 164/255, blue: 85/255, alpha: 1)
        segmentedControl6.highlightColor = UIColor(red: 41/255, green: 197/255, blue: 102/255, alpha: 1)
        segmentedControl6.tint = UIColor.whiteColor()
        self.view.addSubview(segmentedControl6)
        
        // SegmentedControl7
        
        let icons4:[UIImage] = [UIImage(named: "icon2")!, UIImage(named: "icon3")!, UIImage(named: "icon4")!, UIImage(named: "icon5")!]
        let segmentedControl7 = XMSegmentedControl(frame: CGRect(x: 0, y: 364, width: self.view.frame.width, height: 44), segmentIcon: icons4, selectedItemHighlightStyle: XMSelectedItemHighlightStyle.Background)
        segmentedControl7.backgroundColor = UIColor(red: 39/255, green: 54/255, blue: 70/255, alpha: 1)
        segmentedControl7.highlightColor = UIColor(red: 118/255, green: 201/255, blue: 220/255, alpha: 1)
        segmentedControl7.tint = UIColor.whiteColor()
        segmentedControl7.highlightTint = UIColor(red: 39/255, green: 54/255, blue: 70/255, alpha: 1)
        self.view.addSubview(segmentedControl7)
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func xmSegmentedControl(xmSegmentedControl: XMSegmentedControl, selectedSegment: Int) {
        if xmSegmentedControl == segmentedControl1 {
            print("SegmentedControl1 Selected Segment: \(selectedSegment)")
        }
    }
}

